src/server/models/invite-codes.js

```diff
@@ -3,14 +3,11 @@
 const mongoose = require("mongoose")
 const util = require("../util");
 
-// types: { teacher:1, ta:2, student:3 }
 const inviteCodeSchema = new mongoose.Schema({
-    classroom: { type: "ObjectId", index: true, ref: "Classroom" },
-    code: { type: String,  lowercase: true, default: getCode, index: true },
-    type: { type: Number }
+    code: { type: String,  default: getCode, required: true, unique:true },
 }, {
-    timestamps: true
-})
+  timestamps: true,
+});
 
 class InviteCode {
```

Comment:
You can keep the class stuff if you want but since we never query codes again it is unused